### ISSUE TITLE
[BUG] Pool allocator causes segfault in gdf_order_by

### ISSUE THREAD

**Issue by kkraus14** [rapidsai/cudf#792]
In the E2E workflow in https://github.com/rapidsai/notebooks, stacktrace:
```
#0  0x00007fffe0670480 in cuEGLApiInit () from /usr/lib/x86_64-linux-gnu/libcuda.so
#1  0x00007fffe06705a0 in cuEGLApiInit () from /usr/lib/x86_64-linux-gnu/libcuda.so
#2  0x00007fffe0588555 in cuMemGetAttribute_v2 () from /usr/lib/x86_64-linux-gnu/libcuda.so
#3  0x00007fffe06cf83f in cuStreamGetFlags () from /usr/lib/x86_64-linux-gnu/libcuda.so
#4  0x00007fffdf6e9ebf in ?? () from /usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart.so.9.2
#5  0x00007fffdf71231f in cudaStreamGetFlags () from /usr/local/cuda-9.2/targets/x86_64-linux/lib/libcudart.so.9.2
#6  0x00007fffdf964d7f in cnmem::Manager::setStream (this=0x227c3d90, stream=0x7ffb5a5af200)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/thirdparty/cnmem/src/cnmem.cpp:392
#7  0x00007fffdf964384 in cnmemRegisterStream (stream=0x7ffb5a5af200)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/thirdparty/cnmem/src/cnmem.cpp:1166
#8  0x00007fffdf95ef14 in rmm::Manager::registerStream (this=0x7fffdfb6e160 <rmm::Manager::getInstance()::instance>, 
    stream=0x7ffb5a5af200) at /home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/src/memory_manager.cpp:93
#9  0x00007fffc883c27f in rmm::alloc<void> (ptr=0x7fffffffb980, size=3554047, stream=0x7ffb5a5af200, 
    file=0x7fffc8f217c0 <_ZN3rmmL17RMM_USAGE_LOGGINGE+3889> "/home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/include/rmm/thrust_rmm_allocator.h", line=48) at /home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/include/rmm/rmm.hpp:133
#10 0x00007fffc889891a in rmm_allocator<char>::allocate (this=0x7fffffffc1b0, n=3554047)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/thirdparty/rmm/include/rmm/thrust_rmm_allocator.h:48
#11 0x00007fffc88984c1 in thrust::detail::allocator_traits<rmm_allocator<char> >::allocate(rmm_allocator<char>&, unsigned long)::workaround_warnings::allocate(rmm_allocator<char>&, unsigned long) (a=..., n=3554047)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/allocator/allocator_traits.inl:230
#12 0x00007fffc88984e7 in thrust::detail::allocator_traits<rmm_allocator<char> >::allocate (a=..., n=3554047)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/allocator/allocator_traits.inl:234
#13 0x00007fffc8896fb7 in thrust::detail::get_temporary_buffer<char, rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base> (system=..., n=3554047) at /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/execute_with_allocator.h:86
#14 0x00007fffc88953da in thrust::get_temporary_buffer<char, thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base> > (exec=..., n=3554047)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/temporary_buffer.h:62
#15 0x00007fffc8894979 in thrust::cuda_cub::get_memory_buffer<thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base> > (policy=..., n=3554047)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/system/cuda/detail/memory_buffer.h:57
#16 0x00007fffc889383a in thrust::cuda_cub::__merge_sort::merge_sort<thrust::detail::integral_constant<bool, false>, thrust::detail::integral_constant<bool, false>, thrust::cuda_cub::execution_policy<thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base> >, int*, int*, __nv_dl_wrapper_t<__nv_dl_tag<void (*)(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*), &(void multi_col_sort<int>(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*)), 2u>, LesserRTTI<int> > > (compare_op=..., items_first=0x0, keys_last=0x7ffb5a91244c, keys_first=0x7ffb5a5af200, policy=...)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/system/cuda/detail/sort.h:1336
#17 thrust::cuda_cub::__smart_sort::smart_sort<thrust::detail::integral_constant<bool, false>, thrust::detail::integral_constant<bool, false>, thrust::cuda_cub::execution_policy<thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base> >, int*, int*, __nv_dl_wrapper_t<__nv_dl_tag<void (*)(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*), &(void multi_col_sort<int>(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*)), 2u>, LesserRTTI<int> > > (
    compare_op=..., items_first=0x0, keys_last=0x7ffb5a91244c, keys_first=0x7ffb5a5af200, policy=...)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/system/cuda/detail/sort.h:1576
#18 thrust::cuda_cub::sort<thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base>, int*, __nv_dl_wrapper_t<__nv_dl_tag<void (*)(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*), &(void multi_col_sort<int>(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*)), 2u>, LesserRTTI<int> > > (policy=..., first=0x7ffb5a5af200, 
    last=0x7ffb5a91244c, compare_op=...)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/system/cuda/detail/sort.h:1653
#19 0x00007fffc8890f2d in thrust::sort<thrust::detail::execute_with_allocator<rmm_allocator<char>, thrust::cuda_cub::execute_on_stream_base>, int*, __nv_dl_wrapper_t<__nv_dl_tag<void (*)(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*), &(void multi_col_sort<int>(void* const*, unsigned char* const*, int*, signed char*, unsigned long, unsigned long, bool, int*, bool, CUstream_st*)), 2u>, LesserRTTI<int> > > (exec=..., 
    first=0x7ffb5a5af200, last=0x7ffb5a91244c, comp=...)
    at /usr/local/cuda/targets/x86_64-linux/include/thrust/detail/sort.inl:56
#20 0x00007fffc889067c in multi_col_sort<int> (d_cols=0x7ffb45b73c00, d_valids=0x7ffb45b73e00, d_col_types=0x7ffb45b74000, 
    d_asc_desc=0x0, ncols=2, nrows=887955, have_nulls=false, d_indx=0x7ffb5a5af200, nulls_are_smallest=false, stream=0x0)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/src/orderby/../sqls/sqls_rtti_comp.h:814
#21 0x00007fffc888d6b7 in (anonymous namespace)::multi_col_order_by (cols=0x7fffffffc910, asc_desc=0x0, ncols=2, 
    output_indices=0x7fffffffc570, flag_nulls_are_smallest=false)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/src/orderby/orderby.cu:57
#22 0x00007fffc888d77f in gdf_order_by (cols=0x7fffffffc910, asc_desc=0x0, ncols=2, output_indices=0x7fffffffc570, 
    flag_nulls_are_smallest=0) at /home/jhemstad/RAPIDS/repro/cudf/cpp/src/orderby/orderby.cu:88
#23 0x00007fffc8b281c4 in GroupbyHash<double, int, min_op<double> > (groupby_input_table=..., 
    in_aggregation_column=0x7ffb5cc71e00, groupby_output_table=..., out_aggregation_column=0x7ffb7cf6e200, 
    out_size=0x7fffffffc620, aggregation_op=..., sort_result=true)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/src/groupby/groupby_compute_api.h:220
#24 0x00007fffc8b1fb11 in typed_groupby<double, min_op, int> (groupby_input_table=..., in_aggregation_column=0x22294050, 
    groupby_output_table=..., out_aggregation_column=0x22a61ce0, sort_result=true)
    at /home/jhemstad/RAPIDS/repro/cudf/cpp/src/groupby/hash_groupby.cuh:63
```

**Comment by kkraus14** [comment]
cc @harrism @jrhemstad

**Comment by kkraus14** [comment]
Simple Python repro:
```python
from librmm_cffi import librmm_config as rmm_cfg
rmm_cfg.use_pool_allocator = True
import cudf
cudf._gdf.rmm_initialize()

df = cudf.DataFrame()
df['a'] = [1,2,3,4,5]
df['b'] = [5,4,3,2,1]
print(df.sort_values(['a']))
```

**Comment by kkraus14** [comment]
Resolved by #794.
